tests/e2e/specs/backend/product-query/atomic-blocks.test.ts

```diff
@@ -0,0 +1,127 @@
+/**
+ * External dependencies
+ */
+import { canvas, setPostContent, insertBlock } from '@wordpress/e2e-test-utils';
+import { visitBlockPage, saveOrPublish } from '@woocommerce/blocks-test-utils';
+
+/**
+ * Internal dependencies
+ */
+import { GUTENBERG_EDITOR_CONTEXT, describeOrSkip } from '../../../utils';
+import { shopper } from '../../../../utils';
+
+const block = {
+	name: 'Product Query',
+	slug: 'woocommerce/product-query',
+	class: '.wp-block-query',
+};
+
+describeOrSkip( GUTENBERG_EDITOR_CONTEXT === 'gutenberg' )(
+	'Product Query > Atomic blocks',
+	() => {
+		beforeEach( async () => {
+			await visitBlockPage( `${ block.name } Block` );
+		} );
+		afterAll( async () => {
+			await visitBlockPage( `${ block.name } Block` );
+			await setPostContent( '' );
+			await insertBlock( 'Product Query' );
+			await saveOrPublish();
+		} );
+
+		it( 'Can add the Add to Cart Button block and render it on the front end.', async () => {
+			const canvasEl = canvas();
+			await canvasEl.waitForSelector(
+				`${ block.class } .wp-block-woocommerce-product-image a img`
```

Comment:
Please try to avoid as much as possible the use of [magic strings](https://deviq.com/antipatterns/magic-strings) as selectors. You can scope your selectors within your `block` object above, like was, for example, done by Luigi in `filter-products-by-attribute.test.ts` among others.